source/applications/advanced/color_balance.py

```diff
@@ -0,0 +1,203 @@
+"""
+Balance color for RGB image.
+"""
+
+import datetime
+import numpy as np
+import matplotlib.pyplot as plt
+import zivid
+
+
+def _display_rgb(rgb, title):
+    """Display RGB image.
+
+    Args:
+        rgb: RGB image
+        title: Image title
+
+    Returns None
+
+    """
+    plt.figure()
+    plt.imshow(rgb)
+    plt.title(title)
+    plt.show(block=False)
+
+
+def _set_settings(dimension, iris, exposure_time, brightness, gain):
+    """Set settings for capture (3D or 2D).
+
+    Args:
+        dimension: '3d' or '2d'
+        iris: Iris
+        exposure_time: Exposure time
+        brightness: Projector brightness
+        gain: Gain
+
+    Returns:
+        settings: Capture settings (3D or 2D)
+
+    """
+    if dimension == "3d":
+        settings = zivid.Settings()
+        settings.iris = iris
+        settings.exposure_time = datetime.timedelta(microseconds=exposure_time)
+        settings.brightness = brightness
+        settings.gain = gain
+    if dimension == "2d":
+        settings = zivid.Settings2D()
+        settings.iris = iris
+        settings.exposure_time = datetime.timedelta(microseconds=exposure_time)
+        settings.brightness = brightness
+        settings.gain = gain
+
+    return settings
+
+
+def _capture_rgb(camera, settings_2d):
+    """Capture 2D RGB image.
+
+    Args:
+        camera: Zivid camera
+        settings_2d: 2D capture settings
+
+    Returns:
+        rgb: RGB image
+
+    """
+    frame_2d = camera.capture_2d(settings_2d)
+    image = frame_2d.image().to_array()
+    rgb = np.dstack([image["r"], image["g"], image["b"], image["a"]])
+
+    return rgb
+
+
+def _compute_mean_rgb(rgb, pixels):
+    """Compute mean RGB values.
+
+    Args:
+        rgb: RGB image
+        pixels: Number of central pixels for computation
+
+    Returns:
+        mean_red: Red channel mean value
+        mean_green: Green channel mean value
+        mean_blue: Blue channel mean value
+
+    """
+    height = np.shape(rgb)[0]
+    width = np.shape(rgb)[1]
+    red = rgb[
+        int((height - pixels) / 2) : int((height + pixels) / 2),
+        int((width - pixels) / 2) : int((width + pixels) / 2),
+        0,
+    ]
+    green = rgb[
+        int((height - pixels) / 2) : int((height + pixels) / 2),
+        int((width - pixels) / 2) : int((width + pixels) / 2),
+        1,
+    ]
+    blue = rgb[
+        int((height - pixels) / 2) : int((height + pixels) / 2),
+        int((width - pixels) / 2) : int((width + pixels) / 2),
+        2,
+    ]
+    mean_red = np.mean(np.reshape(red, -1, []), dtype=np.float64)
```

Comment:
What is the last argument in `np.reshape(red, -1, [])` ?

Reply:
It is the order, which doesn't matter.

Reply:
I can also remove it `np.reshape(red, -1)` it would work the same

Reply:
Is it a required argument? Or could you drop it altogether?